### DISCUSSION TITLE
Any way to force YAML value as quoted string?

### DISCUSSION THREAD

**Discussion by djarnis73** [apple/pkl#829] [Q&A] [answered]
I have run into an issue where I'm generating code for some kubernetes deployment that uses datadog integration for creating datadog monitors.

It has a requirement that some values are quoted as string even though they are numbers.

So when I try to deploy I get errors from kubernetes like:
```
error when patching "...": DatadogMonitor.datadoghq.com "my-monitor-name" is invalid: [spec.options.thresholds.critical: Invalid value: "integer": spec.options.thresholds.critical in body must be of type string: "integer", spec.options.thresholds.warning: Invalid value: "integer": spec.options.thresholds.warning in body must be of type string: "integer"]
```
I did try the various compat modes in `YamlRenderer`without success, also tried to see if I could do tricks with a converter and YAML tags like:
```
renderer = new YamlRenderer {
  converters {
    ["options.thresholds.critical"] = (v) -> "!!str " + v.toString()
    ["options.thresholds.warning"] = (v) -> "!!str " + v.toString()
  }
}
```
But that leads to the tag being inside a string, like:
```
options:
  thresholds:
    warning: '!!str 25'
    critical: '!!str 51'
```
instead of:
```
options:
  thresholds:
    warning: !!str 25
    critical: !!str 51
```
Will I have to do post some custom post processing of generated YAML to fix this?

**Comment by HT154** [comment, marked as answer]
The solution here is actually extremely straightforward: If a value is a String in Pkl it will be a string in the resulting YAML!

If all keys under `options.thresholds` should have string values, then you want this:

```pkl
renderer = new YamlRenderer {
  converters {
    ["options.thresholds.*"] = (v) -> v.toString()
  }
}
```

Pkl's YAML renderer knows when to wrap a string value in quotes to avoid a collision with other YAML types (number, boolean, etc.). This is also why the attempt to add `!!str` results in a quoted string.

**Reply by djarnis73** [reply]
Wow, I could have sworn I tried that out already, but apparently not 🤯 .

Thanks
